src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.cs

```diff
@@ -121,5 +132,29 @@ protected override SyntaxToken GetMethodNameAtInvocation(IEnumerable<SyntaxNodeO
 
             return OperationStatus.Succeeded;
         }
+
+        protected override async Task<(Document document, SyntaxToken methodName, SyntaxNode methodDefinition)> InsertNewLineBeforeLocalFunctionIfNecessaryAsync(
+            Document document,
+            SyntaxToken methodName,
+            SyntaxNode methodDefinition,
+            CancellationToken cancellationToken)
+        {
+            // Checking to see if there is already an empty line before the local method declaration.
+            var leadingTrivia = methodDefinition.GetLeadingTrivia();
+            if (!leadingTrivia.Any(t => t.IsKind(SyntaxKind.EndOfLineTrivia)))
+            {
+                var originalMethodDefinition = methodDefinition;
+                methodDefinition = methodDefinition.WithPrependedLeadingTrivia(SpecializedCollections.SingletonEnumerable(SyntaxFactory.CarriageReturnLineFeed));
+
+                // Generating the new document and associated variables.
+                var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                document = document.WithSyntaxRoot(root.ReplaceNode(originalMethodDefinition, methodDefinition));
+
+                var newRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                methodName = newRoot.FindToken(methodName.SpanStart);
```

Comment:
If the methodName is always retrieved at the same start, does it actually change? May be able to remove it from the parameters and return

Reply:
I tried removing methodName from the parameters and it didn't work -- I think it has something to do with needing the exact token with the rename annotation.